### PR TITLE
[MINOR][DOCS] Remove Python 3.9 and Apache Arrow warning comment

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to update the doc of arrow.

### Why are the changes needed?
Our docs (https://spark.apache.org/docs/latest/index.html#downloading):
image
But, PyArrow is currently compatible with Python 3.9.
https://arrow.apache.org/docs/python/install.html#python-compatibility
image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -45,7 +45,6 @@ Java 8 prior to version 8u201 support is deprecated as of Spark 3.2.0.
 When using the Scala API, it is necessary for applications to use the same version of Scala that Spark was compiled for.
 For example, when using Scala 2.13, use Spark compiled for 2.13, and compile code/applications for Scala 2.13 as well.
 
-For Python 3.9, Arrow optimization and pandas UDFs might not work due to the supported Python versions in Apache Arrow. Please refer to the latest [Python Compatibility](https://arrow.apache.org/docs/python/install.html#python-compatibility) page.
```
Comment (dongjoon-hyun) [inline docs/index.md:48]: +1 for the removal.

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -34,13 +34,13 @@ Please see [Spark Security](security.html) and the specific security sections in
...
+Alternatively the [Pod Template](#pod-template) feature can be used to add a [Security Context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#volumes-and-file-systems) with a `runAsUser` to the pods that Spark submits.  This can be used to override the `USER` directives in the images themselves.  Please bear in mind that this requires cooperation from your users and as such may not be a suitable solution for shared environments.  Cluster administrators should use [Pod Security Admission](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod) if they wish to limit the users that pods may run as.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:37]: I'm not sure about this part because K8s 1.25 is not available in EKS environment yet.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Could you narrow down to Arrow only? For K8s, it looks like controversial because K8s v1.25 is not available yet and the proposed one is still beta until v1.24.

**Comment by panbingkun** [comment]
> it looks like controversial because K8s v1.25 is not available yet and the proposed one is still beta until v1.24.

Ok, let me narrow down to Arrow first.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
Thank you, @panbingkun .

**Pull request closed without merging**